components/eventing-controller/controllers/subscription/nats/unit_test.go

```diff
@@ -0,0 +1,309 @@
+package nats
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/mocks"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestHandleSubscriptionDeletion(t *testing.T) {
+	g := NewGomegaWithT(t)
+	testEnvironment := setupTestEnvironment(t)
+	ctx, r, mockedBackend := testEnvironment.Context, testEnvironment.Reconciler, testEnvironment.Backend
+
+	subscription := &eventingv1alpha1.Subscription{
+		ObjectMeta: v1.ObjectMeta{
+			Name:       "test",
+			Namespace:  namespaceName,
+			Finalizers: []string{}, // empty finalizers
+		},
+	}
+
+	err := testEnvironment.Reconciler.Client.Create(testEnvironment.Context, subscription)
+	g.Expect(err).Should(BeNil())
```

Comment:
The fake client provides the possibility to set the initial objects:
```go
// source: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake
client := NewFakeClientWithScheme(scheme, initObjs...) // initObjs is a slice of runtime.Object
```

We could use this instead of calling `Create` on the client. It gives the reader an initial feeling which objects are present in the `fake cluster`.

Which version do you prefer ?

```suggestion
	subscription := &eventingv1alpha1.Subscription{
		ObjectMeta: v1.ObjectMeta{
			Name:       "test",
			Namespace:  namespaceName,
			Finalizers: []string{}, // empty finalizers
		},
	}
	testEnvironment := setupTestEnvironment(t, 
	    WithInitialObjects([]runtime.object{subscription}) // initial cluster state
	 )
	ctx, r, mockedBackend := testEnvironment.Context, testEnvironment.Reconciler, testEnvironment.Backend
```